favorites.py

```diff
@@ -260,41 +280,54 @@ def add_group(self, s):
     def set(self, s, group_name=None):
         """Add file in global or group list."""
 
+        s = {"file": s, "alias": basename(s)}
+
         if group_name is None:
             self.obj.files["files"].append(s)
         else:
             self.obj.files["groups"][group_name].append(s)
 
-    def exists(self, s, group=False, group_name=None):
-        """Check if group or file exists."""
+    def group_exists(self, s):
+        """Check if group exists."""
 
-        if group:
-            # See if group exists
-            return True if s in self.obj.files["groups"] else False
+        return s in self.obj.files["groups"]
+
+    def file_index(self, s, group_name=None):
+        """Check if file exists, and return its index in this case."""
+
+        def is_present_in(files_list):
+            """Returns the index if the file is present"""

```

Comment:
First line of doc string should have a period: ./favorites.py:298:1: D400 First line should end with a period.